iotaledger/iri
# Will fix #1231: ZMQ publishes to TCP and IPC instead of just one of them

# Description

This change makes it possible to run ZMQ on IPC or/and TCP socket:

`--zmq-enable-tcp true` - will run ZMQ on TCP default port 5556
`--zmq-enable-ipc true` - will run ZMQ on IPC default socket ipc://iri
`--zmq-enabled true` - will run ZMQ as before on TCP and IPC socket for backward compatibility (should be removed in the future)

Fixes #1231: ZMQ publishes to TCP and IPC instead of just one of them

## Type of change

- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?
Implemented new unit tests ZMQConfigTest.java for testing command line parameters

- start iri with `--zmq-enable-tcp true` parameter and check ZMQ binds to TCP port 5556
- start iri with `--zmq-enable-ipc true` parameter and check ZMQ binds to ips socket ipc://iri
- start iri with `--zmq-enabled true` parameter and check ZMQ binds like before change

# Checklist:

_Please delete items that are not relevant._
